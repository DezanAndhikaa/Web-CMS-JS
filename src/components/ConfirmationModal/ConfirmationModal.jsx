

import React from 'react';
import { DialogContent, Button, Modal } from '@material-ui/core';
import { ImgApproved, DelSuccess, Success } from '../../assets/imgs';
import './ConfirmationModal.scss';
import CloseNotif from '../CloseNotif/CloseNotif';

export default class ConfirmationModal extends React.PureComponent {
  render() {
    if(this.props.idModal === "Approved"){
      return (
        <Modal open={this.props.openModal} onClose={this.props.onClose} className="modal-container">
            <DialogContent className="confirmation-modal-content">
              <div className="confirmation-modal">
              <CloseNotif onClose={this.props.onClose}/>
                  <div className="confirmation-container">
                    <p className="confirmation-title">Successful</p>
                    <img className="image-approved" src={ImgApproved} alt="" />
                    <p className="confirmation-caption">Please resume your activities</p>
                    <br></br>
                    <Button className="btn-ok" onClick={ () => {this.props.onClose()} }>Continue</Button>
                  </div>
              </div>
            </DialogContent>
            </Modal>
        );
    }else if(this.props.idModal === "SAP"){
      return (
        <Modal open={this.props.isModal} onClose={this.props.onClose} className="modal-container">
            <DialogContent className="confirmation-modal-content">
              <div className="confirmation-modal">
                  <div className="confirmation-container">
                    <p className="confirmation-title">Successful</p>
                    <img className="confirmation-image" src={Success} alt="" />
                    <p className="confirmation-caption">You can see the data SAP Issue</p>
                    <p className="confirmation-caption">to the <b>Status</b></p>
                    <br></br>
                    <Button className="btn-ok" onClick={ () => {this.props.isModalClosed();this.props.handleReload()} }>Continue</Button>
                  </div>
              </div>
            </DialogContent>
            </Modal>
        );
    } else if(this.props.idModal === "SAP-Failed"){
      return (
        <Modal open={this.props.isModal} onClose={this.props.isModalClosed} className="modal-container">
				  <DialogContent className="confirmation-modal-content">
            <div className="confirmation-modal">
              <CloseNotif onClose={this.props.isModalClosed}/>
              <div className="confirmation-container">
                <p className="confirmation-title">You have not completed</p>
                <p className="confirmation-title">the SAP Issue form</p>
                <img className="confirmation-image" src={DelSuccess} alt="" />
                <p className="confirmation-caption">Do you want to continue?</p>
                <div className="btn-container">
                  {this.props.idFailed === "CloseBtn" ? 
                    <Button className="btn-tidak" onClick={ () => {this.props.isModalClosed();this.props.handleReload()} }>No</Button>
                  : 
                    <Button className="btn-tidak" onClick={this.props.backToConfirmModal}>No</Button>
                  }
                  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<Button className="btn-ya" onClick={this.props.backToSAP}>Yes</Button>
                </div>
              </div>
            </div>
          </DialogContent>
        </Modal>
      );
    } else if(this.props.idModal === "Delete Success"){
      return (
        <Modal open={this.props.openModal} onClose={this.props.onClose} className="modal-container">
          <DialogContent className="confirmation-modal-content">
            <div className="confirmation-modal">
              <div className="confirmation-container">
                  <p className="confirmation-title">Successful</p>
                  <img className="confirmation-image" src={Success} alt="" /> 
                  <p className="confirmation-caption">The data was deleted</p>
                  <p className="confirmation-caption">successfully.</p>
                  <br></br>
                  <Button className="btn-ok" onClick={ () => {this.props.onClose()} }>Continue</Button>
              </div>
            </div>
          </DialogContent>
        </Modal>
      );
    } else if(this.props.idModal === "Revised"){
      return (
        <Modal open={this.props.openModal} onClose={this.props.onClose} className="modal-container">
          <DialogContent className="confirmation-modal-content">
            <div className="confirmation-modal">
              <div className="confirmation-container">
                  <p className="confirmation-title">Successful</p>
                  <img className="confirmation-image" src={Success} alt="" /> 
                  <p className="confirmation-caption">The data was revised</p>
                  <p className="confirmation-caption">successfully.</p>
                  <br></br>
                  <Button className="btn-ok" onClick={ () => {this.props.onClose()} }>Continue</Button>
              </div>
            </div>
          </DialogContent>
        </Modal>
      );
    }
  }
}