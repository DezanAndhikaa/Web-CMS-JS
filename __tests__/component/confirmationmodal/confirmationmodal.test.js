import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import ConfirmationModal from 'components/ConfirmationModal/ConfirmationModal';

Enzyme.configure({adapter:new Adapter()});
it("render confirmation modal correctly", () => {
    const wrapper = shallow (
        <ConfirmationModal />
    )
    expect(wrapper).toMatchSnapshot();
})