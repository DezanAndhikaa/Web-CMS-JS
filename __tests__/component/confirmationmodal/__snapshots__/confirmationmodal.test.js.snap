// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render confirmation modal correctly 1`] = `""`;
